crates/blockchain-tree/src/blockchain_tree.rs

```diff
@@ -810,6 +804,49 @@ impl<DB: Database, C: Consensus, EF: ExecutorFactory> BlockchainTree<DB, C, EF>
         }
     }
 
+    /// Attempts to re-insert _all_ child blocks on top of tracked chains.
+    ///
+    /// This will return the outcome of all reinsert attempts.
+    pub fn append_buffered_block_all(&mut self) -> Vec<Result<BlockNumHash, InsertBlockError>> {
+        let mut all = Vec::new();
+        while let Some(res) = self.append_buffered_block_one() {
+            all.push(res);
+        }
+        all
+    }
+
+    /// Attempts to re-insert the child blocks on top of tracked chains.
+    ///
+    /// This will return the outcome of the first reinsert attempt.
+    pub fn append_buffered_block_one(&mut self) -> Option<Result<BlockNumHash, InsertBlockError>> {
+        let (chain_id, num_hash) = self.find_appendable_buffered_block()?;
```

Comment:
wdym by fork here? forking what and/or where?


what this does is effectively the same as:

https://github.com/paradigmxyz/reth/blob/bb57556a6ab84f1c9f09eebd1f618220430b5101/crates/blockchain-tree/src/blockchain_tree.rs#L301-L301

Reply:
What if you have the buffer of.
```
        b3
        |
b1    b2
|     /
|  /
[was unknown parent] but it is now `c2`
```

And you want to append those blocks onto the chain.
What is done in `find_appendable_buffered_block_for_chain_tip ` is that it checks only the TIP `let chain_tip = chain.tip().hash();` of the chain.

If we have chain with blocks `[c1, c2]`

first call of `append_buffered_block_one` would extend chain to `[c1,c2,b1]`

The second call would need to create a new chain `[b2]` (the first chain is `[c1,c2,b1]` ), this will not be done as we just check if the block extends the chain (we are checking tip)

Reply:
hmm, not sure I understand this fully, I'm still lacking some understanding of how the tree works.

But I want to get rid of expensive connect buffered.

what would be the best approach?

Reply:
Could a simple solution be to return the lowest block whose parent exists in the tree or canonical chain?